radareorg/cutter
# Implement Editing and Viewing of Types in the Types Widget

**Detailed description**

This pull request implement an Editing and Viewing options for the Types widget. It also renames the class to a more meaningful name.


**Test plan (required)**

1. Open a binary in Cutter
2. Open the Types widget
3. Right Click in the widget and choose Load New Types
4. Insert some Type definitions and press OK. For example:
```
struct abc { char b; };
enum Foo {COW=1,BAR=2};
union fffff { char b; };
```
5. Right Click one of the new Types you added and choose View Type
6. Notice a dialog that opens in a Read-Only mode.
![image](https://user-images.githubusercontent.com/20182642/59128787-b2788d00-8973-11e9-8b69-5ab35e963de1.png)

7. Click OK
8. Right Click one of the new Types you added and choose Edit Type
9. Notice a dialog that opens and allows you to modify the Type.
>

**Closing issues**

closes #1207 , closes #1208
